lib/signatures.ts

```diff
@@ -12,32 +12,31 @@ export async function getSignaturesByAlgorithm(
   filePaths: string[],
   hashType: SignatureHashAlgorithm = 'dubhash',
 ): Promise<(SignatureResult | null)[]> {
-  if (hashType !== 'dubhash' && hashType !== 'uhash') {
+  if (!isHashValidType(hashType)) {
     throw new Error(`Unsupported hashType ${hashType}`);
   }
-  let signatureMapperToBeUsed = getDubHashSignatureMapper(baseDir);
-  if (hashType === 'uhash') {
-    signatureMapperToBeUsed = getUHashSignatureMapper(baseDir);
-  }
-  return await pMap(filePaths, signatureMapperToBeUsed, {
+
+  const signatureMapper = getSignatureMapper(baseDir, hashType);
+
+  return await pMap(filePaths, signatureMapper, {
     concurrency: 20,
   });
 }
 
-const getDubHashSignatureMapper = (baseDir: string) =>
-  async function(filePath: string): Promise<SignatureResult | null> {
-    const absolutePath = join(baseDir, filePath);
-    const fileContents = await readFile(absolutePath);
+function isHashValidType(hashType: string): boolean {
+  return hashType === 'dubhash' || hashType === 'uhash';
```

Comment:
as mentioned in another comment, it was moved to make it more visible for other developers that it might impact them. But since we are going to remove the option it's not a problem in the end.

Reply:
this is not part of the code right now, it's in another "refactoring PR".